### DISCUSSION TITLE
Should we switch to using `flake8 --diff` instead of `flakeheaven`?

### DISCUSSION THREAD

**Discussion by aucampia** [RDFLib/rdflib#2215] [Development]
[flakeheaven](https://flakeheaven.readthedocs.io/en/latest/) is stuck on a rather old version of flake8. `flake8 --diff` is not quite as nice in principle, but the benefit is that it is part of flake8, also it makes it easier for us to add additional flake8 linters without having to fix all existing code. One downside that running flake8 without `--diff` will still be very noisy,.

I think we should move though, unless there is an alternative to flakeheaven that allows baselineing.

**Comment by aucampia** [comment]
Flake8 diff is deprecated, but we should move away from flakeheaven as that likely won't get updated to a newer flake8.

**Comment by jclerman** [comment]
Sometime in the future, another option might be to switch to `ruff`.  That project doesn't support baselines either, but an issue for that has been opened and has generated robust discussion and maintainer interest: https://github.com/charliermarsh/ruff/issues/1149.
